src/features/Layout/components/parts/DropdownListComponent.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { OptionList } from '../../../models';
+import { useAppSelector, useAppDispatch } from '../../../../app/hooks';
+import { getPage, getCharacterList, resetPageNumber, resetSearch} from '../../../redux-slices/characterSlice';
+import {
+    selectStatusAlive, 
+    selectStatusDead,
+    selectStatusUnknown,
+    selectGenderFemale,
+    selectGenderMale,  
+    selectGenderGenderless,
+    selectGenderUnknown,
+    selectName,
+    selectSpecies,
+    selectType,
+} from '../../../redux-slices/dropdownSlice';
+import { 
+    DropdownTitle,
+    DropdownBox,
+    DropdownBoxItem,
+    ResetButton
+} from '../../../appStyles';
+import { generateUrl } from '../../../utils';
+
+
+interface Props {
+    className?: string;
+}
+
+const DropdownListComponent = ({className}: Props) =>
+{
+    const list: OptionList = useAppSelector( state => state.dropdown.optionList);
+    const dispatch = useAppDispatch();
+    const optionList = useAppSelector(state => state.dropdown.optionList);
+    const search = useAppSelector(state => state.character.search);
+
+    const handleInputChange = (action: string) =>
+    {
+        const url = generateUrl(search,optionList);
+        switch(action) 
+        {
+            case 'alive':
+                dispatch(selectStatusAlive()); 
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'dead':
+                dispatch(selectStatusDead());
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'sUnknown':
+                dispatch(selectStatusUnknown())
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'female':
+                dispatch(selectGenderFemale());
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'male':
+                dispatch(selectGenderMale());
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'genderless':
+                dispatch(selectGenderGenderless());
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'gUnknown':
+                dispatch(selectGenderUnknown());
+                dispatch(getPage(url));
+                dispatch(resetPageNumber());
+                break;
+            case 'name':
+                dispatch(selectName());
```

Comment:
Dispatching multiple actions at once can lead to unpredictable results. And can cause unnecessary render cycles.